test/test_suite.py

```diff
@@ -191,3 +205,178 @@ def test_select_exists_false(self, connection):
+            Overriding the tests and add a manual primary key value to avoid the same failures.
+
+            Spanner convert timestamp into `%Y-%m-%dT%H:%M:%S.%fZ` format, so avoid assert
+            failures convert datetime input to the desire timestamp format.
+        """
+        date_table = self.tables.datetime_table
+        config.db.execute(date_table.insert(), {"id": 1, "date_data": self.data})
+
+        row = config.db.execute(select([date_table.c.date_data])).first()
+        compare = self.compare or self.data
+        compare = compare.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+        eq_(row[0].rfc3339(), compare)
+        assert isinstance(row[0], DatetimeWithNanoseconds)
+
+    @requires.standalone_null_binds_whereclause
+    def test_null_bound_comparison(self):
+        """Cloud Spanner supports tables with empty primary key, but
+            only single one row can be inserted into such a table -
+            following insertions will fail with `400 id must not be NULL in table date_table`.
+            Overriding the tests and add a manual primary key value to avoid the same failures.
+        """
+        # this test is based on an Oracle issue observed in #4886.
+        # passing NULL for an expression that needs to be interpreted as
+        # a certain type, does the DBAPI have the info it needs to do this.
+        date_table = self.tables.datetime_table
+        with config.db.connect() as conn:
+            result = conn.execute(
+                date_table.insert(), {"id": 1, "date_data": self.data}
+            )
+            id_ = result.inserted_primary_key[0]
+            stmt = select([date_table.c.id]).where(
+                case(
+                    [
+                        (
+                            bindparam("foo", type_=self.datatype) != None,
+                            bindparam("foo", type_=self.datatype),
+                        )
+                    ],
+                    else_=date_table.c.date_data,
+                )
+                == date_table.c.date_data
+            )
+
+            row = conn.execute(stmt, {"foo": None}).first()
+            eq_(row[0], id_)
+
+
+class DateTimeTest(_DateTimeTest, DateTimeMicrosecondsTest):
+    """
+    DateTime class tests have same failures same as DateTimeMicrosecondsTest class tests,
+    so to avoid that failures and maintain DRY concept just inherit the class to run tests
+    successfully.
+    """
+
+    pass
+
+
+class TimeTests(_TimeMicrosecondsTest, _TimeTest):
+    @pytest.mark.skip("Spanner doesn't support Time data type.")
+    def test_null(self):
+        pass
+
+    @pytest.mark.skip("Spanner doesn't support Time data type.")
+    def test_round_trip(self):
+        pass
+
+    @pytest.mark.skip("Spanner doesn't support Time data type.")
+    @requires.standalone_null_binds_whereclause
+    def test_null_bound_comparison(self):
+        pass
+
+
+class DateTimeCoercedToDateTimeTest(_DateTimeCoercedToDateTimeTest, TimeTests):
```

Comment:
All the tests being skipped in `TimeTests` are also being skipped here. Having the correct skip reason is important so it can remain its own test.

```suggestion
class DateTimeCoercedToDateTimeTest(_DateTimeCoercedToDateTimeTest):
```

Reply:
Additionally, if all tests in the class are being skipped you can instead apply the skip to the class:
```
@pytest.mark.skip("Spanner doesn't coerce dates from datetime.")
class DateTimeCoercedToDateTimeTest(_DateTimeCoercedToDateTimeTest):
    pass
```